Realm/RLMArray.mm

```diff
@@ -299,6 +299,13 @@ - (RLMResults *)objectsWhere:(NSString *)predicateFormat args:(va_list)args
     return [self objectsWithPredicate:[NSPredicate predicateWithFormat:predicateFormat arguments:args]];
 }
 
+- (id)valueForKeyPath:(NSString *)keyPath {
+    if (_backingArray) {
+        return [_backingArray valueForKeyPath:keyPath];
```

Comment:
One problem with delegating to `-[NSArray valueForKeyPath:]` like this is that it will result in standalone arrays supporting more of KVC than persisted arrays (since we reject nested key paths).